aten/src/ATen/native/Linear.cpp

```diff
@@ -92,9 +93,13 @@ Tensor linear(const Tensor& input, const Tensor& weight, const std::optional<Ten
   if (bias->defined() && !input.is_xla()) {
     // Also hit the fused path for contiguous 3D input, if not using xla
     // backend. Reshaping/flattening has some performance implications on xla.
-    if (input.is_contiguous() && input_dim == 3) {
+    if (definitely_contiguous(input.sym_sizes(), input.sym_strides(), input.sym_numel()) && input_dim == 3) {
```

Comment:
I see will have to change it again with my change but fine for now. I will handle it once i rebase.